lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -909,30 +1003,414 @@ class ConvertAtenNativeLayerNormOp
+    MLIRContext *context = op->getContext();
+    Location loc = op->getLoc();
+    Value dY = adaptor.grad_out();
+    Value x = adaptor.input();
+    Value normalizedShape = adaptor.normalized_shape();
+    Value mean = adaptor.mean();
+    Value invSTD = adaptor.rstd();
+    Value weight = adaptor.weight();
+    Value bias = adaptor.bias();
+
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    // TODO: Handle the None cases for the optional parameters:
+    // weight, bias.
+    if (failed(checkNotNone(rewriter, op, dY)) ||
+        failed(checkNotNone(rewriter, op, x)) ||
+        failed(checkNotNone(rewriter, op, mean)) ||
+        failed(checkNotNone(rewriter, op, invSTD)) ||
+        failed(checkNotNone(rewriter, op, weight)) ||
+        failed(checkNotNone(rewriter, op, bias)))
+      return failure();
+
+    SmallVector<bool> outputMask;
+    if (!matchPattern(op.output_mask(), m_TorchConstantBoolList(outputMask)))
+      return rewriter.notifyMatchFailure(op,
+                                         "output masks must be constant bools");
+    assert(outputMask.size() == 3 && "output mask must have only 3 values");
+
+    // TODO: Handle false values for output mask.
+    if (!outputMask[0] || !outputMask[1] || !outputMask[2]) {
+      return rewriter.notifyMatchFailure(
+          op, "false values for output mask is not supported");
+    }
+
+    auto xType = x.getType().cast<RankedTensorType>();
+    auto meanType = mean.getType().cast<RankedTensorType>();
+    auto invSTDType = invSTD.getType().cast<RankedTensorType>();
+    auto weightType = weight.getType().cast<RankedTensorType>();
+    auto biasType = bias.getType().cast<RankedTensorType>();
+    int64_t xRank = xType.getRank();
+    Type elemTy = xType.getElementType();
+
+    // Check if all the arguments meet the requirements.
+    SmallVector<Value> normalizedShapeSizesTorchInt;
+    if (!getListConstructElements(normalizedShape,
+                                  normalizedShapeSizesTorchInt)) {
+      return rewriter.notifyMatchFailure(op,
+                                         "Unimplemented normalized_shape not"
+                                         "constructed from ListConstruct");
+    }
+    SmallVector<Value> normalizedShapeSizesInt = getTypeConvertedValues(
+        rewriter, loc, getTypeConverter(), normalizedShapeSizesTorchInt);
+    int64_t normalizedShapeRank = normalizedShapeSizesInt.size();
+    int64_t axis = xRank - normalizedShapeRank;
+    if (weightType.getRank() != normalizedShapeRank ||
+        biasType.getRank() != normalizedShapeRank ||
+        meanType.getRank() != xRank || invSTDType.getRank() != xRank ||
+        xRank < normalizedShapeRank || normalizedShapeRank < 1) {
+      return rewriter.notifyMatchFailure(
+          op, "Input or weight or bias or mean or invSTD shape or"
+              "normalized shape not compatible");
+    }
```

Comment:
Lots of error messages are packed into one. Please divide it as it will help the user to find the error more easily.